#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, bool_t, double_t, int_t

gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, "Whether to apply this plugin or not", True)
gen.add("publish_shadow_objects", bool_t, 0, "Whether to publish for visualization", False)

gen.add("shadow_scan_range", double_t, 0, "Maximum range to check for scan shadows", 3.0, 0, 20)
gen.add("shadow_scan_angular_resolution", double_t, 0, "Angular resolution of rays to check for shadows", 0.02, 0, 1.0)
gen.add("shadow_scan_half_angle", double_t, 0, "Half width of scan to check for shadows", 1.57, 0, 3.14)
gen.add("min_shadow_size", double_t, 0, "Radius of object that can fit in shadow to declare it shadowed", 0.15, 0, 2.0) 
gen.add("sensor_x_offset", double_t, 0, "The offset in the x axis of the sensors", 0.5, -1.5, 1.5) 

exit(gen.generate("costmap_2d", "costmap_2d", "ShadowPlugin"))
